### PR TITLE
test: add comprehensive unit tests across codebase

### PR DESCRIPTION
**Pull request by ajanraj**
## Summary

Add comprehensive unit tests across the codebase to improve test coverage.

## Changes

- **62 files changed** with **8,932 lines** of test code
- Added Vitest configuration and test setup

### Test Coverage Areas

| Area | Tests Added |
|------|-------------|
| Hooks | use-breakpoint, use-document-title |
| Lib utilities | ai-sdk-utils, chat-error-utils, composio-*, connector-utils, create-agent-tool, depth-limiter, error-codes, error-utils, fetch, file-upload-utils, message-utils, model-utils, motion, prompt-config, provider-error-detector, redacted-content-detector, routes, sanitize, theme-config, time-grouping, types, user-utils, utils |
| Config | api-keys, constants, features, models, openrouter, providers, schemas, suggestions, theme, tools, upload, model-providers |
| Middleware | auth |
| Store | editor-store |
| Theme | apply-theme, theme-fonts, theme-preset-helper, theme-presets |
| Tools | search-errors, search-provider-factory, search-types, search |
| Convex lib | rateLimitConstants, sanitization_helper |
| Components | connector_tool_call |

## Verification

- [x] Lint passes (`bun run lint`)
- [x] Typecheck passes (`bun run typecheck`)

---
## Summary by cubic
Added comprehensive unit tests across hooks, libs, config, middleware, store, theme, tools, components, and Convex helpers to raise coverage and catch regressions. Set up Vitest with jsdom and a shared test setup for reliable runs.

- **New Features**
  - Tests cover key areas: hooks, lib utilities, config (models/schemas), middleware (auth), store/editor, theme, tools/search, components, and Convex helpers.
  - Coverage reporting enabled (v8) with text, json, and html outputs.
  - Shared test setup mocks matchMedia, localStorage, and ResizeObserver.
  - Exported toolListSchema, createAgentInputSchema, SubAgentAnalysis, and analyzeSubAgentExecution to support testing.

- **Dependencies**
  - Bumped @tanstack/react-router, router-devtools, router-ssr-query, react-start, and router-plugin to 1.144.0.

<sup>Written for commit 618a1737272bc6977f03746bc14faac62c4d08ee. Summary will update automatically on new commits.</sup>

<h3>Greptile Summary</h3>

Added comprehensive unit tests across the codebase covering hooks, utilities, configuration, middleware, stores, themes, tools, and Convex backend functions, significantly improving test coverage.

## Key Additions

- **Test Infrastructure**: Vitest configuration with jsdom environment, coverage reporting, and test setup with essential browser API mocks
- **62 test files** covering critical functionality across hooks, lib utilities, config modules, middleware, stores, themes, tools, and Convex functions
- **8,932 lines of test code** with thorough coverage of edge cases, error handling, and happy paths

## Test Quality

- Well-structured tests with clear describe/it blocks following Vitest conventions
- Proper use of mocking (vi.fn()) for browser APIs and external dependencies
- Good coverage of edge cases (null/undefined inputs, empty arrays, boundary conditions)
- Appropriate use of fake timers for time-dependent tests
- Cleanup with beforeEach/afterEach hooks

## Issues Found

- `src/lib/__tests__/composio-utils.test.ts` uses `vi.resetModules()` which is not available in Vitest v4/Bun test environment, causing test failures

<h3>Confidence Score: 4/5</h3>

- This PR is largely safe to merge with one test compatibility issue that needs fixing
- Score reflects high-quality test additions with comprehensive coverage, but docked one point due to the `vi.resetModules()` compatibility issue in composio-utils tests that causes test failures in the current Bun/Vitest v4 environment
- Pay attention to `src/lib/__tests__/composio-utils.test.ts` - fix the `vi.resetModules()` call before merging

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| vitest.config.ts | Added comprehensive Vitest configuration with jsdom environment, coverage setup, and proper path resolution |
| src/test/setup.ts | Created test setup with essential browser API mocks (matchMedia, localStorage, ResizeObserver, IntersectionObserver) |
| src/lib/__tests__/composio-utils.test.ts | Tests for Composio auth config handling, but uses `vi.resetModules()` which is not available in Bun/Vitest v4 |
| src/lib/__tests__/ai-sdk-utils.test.ts | Comprehensive tests for AI SDK utilities including attachment handling, reasoning extraction, and file part conversions |
| src/lib/store/__tests__/editor-store.test.ts | Complete coverage of theme editor store including undo/redo, checkpoints, and state management |
| convex/lib/__tests__/sanitization_helper.test.ts | Comprehensive tests for message sanitization including tool redaction and file handling |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Dev as Developer
    participant Vitest as Vitest Test Runner
    participant Setup as Test Setup
    participant Tests as Test Suites
    participant Mocks as Mocked APIs

    Dev->>Vitest: Run `bun test`
    Vitest->>Setup: Execute setup.ts
    Setup->>Mocks: Mock window.matchMedia
    Setup->>Mocks: Mock localStorage
    Setup->>Mocks: Mock ResizeObserver
    Setup->>Mocks: Mock IntersectionObserver
    Setup-->>Vitest: Setup complete
    
    Vitest->>Tests: Execute test files
    
    Tests->>Tests: src/lib/__tests__/ai-sdk-utils.test.ts
    Tests->>Tests: src/hooks/__tests__/use-breakpoint.test.ts
    Tests->>Tests: src/lib/store/__tests__/editor-store.test.ts
    Tests->>Tests: convex/lib/__tests__/sanitization_helper.test.ts
    Tests->>Tests: src/utils/__tests__/time-utils.test.ts
    Tests->>Tests: ...and 57 more test files
    
    Tests->>Mocks: Use mocked APIs
    Mocks-->>Tests: Return mock responses
    
    Tests-->>Vitest: Test results
    Vitest-->>Dev: Coverage report & pass/fail status
```

### CODE REVIEW

File: src/lib/__tests__/composio-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+
+describe("composio-utils", () => {
+	const originalEnv = { ...process.env };
+
+	beforeEach(() => {
+		vi.resetModules();
```
[the remark this answers is not in the document]
Comment (ajanraj) [reply src/lib/__tests__/composio-utils.test.ts:7]: It works
